File: mqlservice/src/main/configuration/keywords/Attribute.java
package configuration.keywords;

import com.google.gson.annotations.SerializedName;

import java.util.HashSet;
import java.util.Set;

public class Attribute {
    @SerializedName("name")
    public String keyword;
    @SerializedName("keywords")
    public Set<String> synonyms = new HashSet<>();
}

File: cms/api/src/keywords/attribute.js
"use strict";

class Attribute {
    constructor(name, synonyms) {
        this.synonyms = synonyms;
        this.name = name;
        this.synonyms.push(name);
    }

    addSynonym(keyword) {
        if (!this.synonyms.contains(keyword)) {
            this.synonyms.push(keyword);
        }
    }


}

module.exports = Attribute;